snuba/request/request_settings.py

```diff
@@ -23,6 +23,10 @@ def get_turbo(self) -> bool:
     def get_consistent(self) -> bool:
         pass
 
+    @abstractmethod
+    def set_consistent(self, consistent: bool) -> None:
+        pass
+
```

Comment:
Can we keep the request settings immutable? That gave us the guarantee that nobody can tweak the settings during the query processing. Which is desirable as the settings are meant to define the environment the query runs into (which is generally driven by the parameters provided by the user).

If this is meant to avoid creating a copy of the request object, I would not be too concerned as you can create a new request object without deepcopying the content. That will be cheap enough.

Reply:
We can, though it's quite a bit more verbose to do so. Also I would still want to deepcopy the entire object to be safe since we were doing this before and I don't think we can confidently drop that as it may be mutated somewhere in the pipeline. Since request.settings is already immutable (it already has `add_rate_limit()`) method, this seems like a cleaner solution to me.

Reply:
Actually, it wasn't too bad - updated